contracts/src/Surge.sol

```diff
@@ -101,87 +70,50 @@ contract Surge is ERC721, ReentrancyGuard, Ownable, ERC721Enumerable {
         external
         payable
         nonReentrant
-        isSaleActive
-        maxMint(_amountOfTokens)
+        verifyMint(_amountOfTokens)
         isEnoughEth(_amountOfTokens)
     {
-        for (uint256 i = 0; i < _amountOfTokens; i++) {
-            uint256 newTokenId = _tokenIds.current() + 1;
-            require(newTokenId <= MAX_TOKENS, "No available tokens to mint");
-            _safeMint(msg.sender, newTokenId);
-            _tokenIds.increment();
-        }
+        require(saleIsActive, "Sale is currently not active");
+        _safeMint(msg.sender, _amountOfTokens);
     }
 
-    //add a wallet to give them access to the presale
-    function addToPresale(address _wallet) external onlyOwner {
-        require(!_presaleApproved[_wallet], "Wallet is already in the presale");
-        _presaleApproved[_wallet] = true;
+    //add wallets to give them access to the presale
+    function addToPresale(address[] calldata _wallet) external onlyOwner {        
```

Comment:
If `wallet` is an `merkleTree` of `wallets`, this should be named plural to avoid confusion

Reply:
agreed, updated!